scout/adapter/mongo/disease_terms.py

```diff
@@ -102,7 +108,7 @@ def disease_terms(
         filter_project: Optional[dict] = DISEASE_FILTER_PROJECT,
     ) -> list:
         """Return all disease terms for a gene HGNC ID. Optionally filter the returned key/values using filter_project. By default do not return disease-associated genes and HPO terms."""
-        query = {"source": "OMIM"}
```

Comment:
I don't like that if one provides an hgnc id then it returns only OMIM terms while without gene it returns all terms. I think it has potential to mess things up. I would fix it in one of these 2 ways:

1 . Split this function into 2 functions:
- disease_terms() -> return all terms
- disease_terms_by_gene (source: Optional[str]=None) -> return terms by gene with the option to specify the source or not

2) add source as optional parameter of the disease_terms() function.

Reply:
Function have been split in two and uses have been updated accordingly